server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1641,13 +1654,28 @@ public BlobContainer shardContainer(IndexId indexId, int shardId) {
      *
      * @param repositorySettings repository settings
      * @param setting            setting to use to configure rate limiter
-     * @return rate limiter or null of no throttling is needed
+     * @param warnIfOverRecovery log a warning if rate limit setting is over the effective recovery rate limit
+     * @return rate limiter or null if no throttling is needed
      */
-    private static RateLimiter getRateLimiter(Settings repositorySettings, Setting<ByteSizeValue> setting) {
+    private RateLimiter getRateLimiter(Settings repositorySettings, Setting<ByteSizeValue> setting, boolean warnIfOverRecovery) {
         ByteSizeValue maxSnapshotBytesPerSec = setting.get(repositorySettings);
         if (maxSnapshotBytesPerSec.getBytes() <= 0) {
             return null;
         } else {
+            if (warnIfOverRecovery && recoverySettings.rateLimiter() != null) {
+                double effectiveRecoverySpeed = recoverySettings.rateLimiter().getMBPerSec();
+                if (maxSnapshotBytesPerSec.getMbFrac() > effectiveRecoverySpeed) {
+                    logger.warn(
+                        "[{}] repository snapshot rate limit [{}={}] will be capped by the effective recovery rate limit [{}] per sec",
```

Comment:
Maybe something like "repository [foo] has a maximum bytes per second limit for snapshot of [max_snapshot_bytes_per_sec=x] which is above the limit allowed for recovery [indices.recovery.max_bytes_per_sec=y], repository limit will be ignored in favor of [y]" ?

Reply:
I modified a bit the warning message by taking into account what you suggested. I liked the term "effective" because it's not only the indices.recovery.max_bytes_per_sec setting that affects but a number of other recovery settings potentially. Please take a look at the new message and give more feedback if/as necessary. Thanks!